File: Benchmarking/src/gen582.swift
  import Dispatch
  @inline(never)
  func f0(_ v0: [[Double]], _ v1: [[Double]], _ v2: [[Double]], _ v3: [[Double]], _ v4: Double) -> Double {
    var v9: Double = v4
    v9 = v4
    var v28: Double = v4
    let v22: [Double] = [v9, v4, v28, v28, v9, v9]
    var v16: [Double] = v22
    let v35: [Double] = v1[1]
    v16[2] = v9
    let v33: Double = v35[0]
    v16 = v22
    v28 = v33
    let v95: [[Double]] = [v22, v16, v22, v16, v22, v16]
    var v219: [[Double]] = v95
    let v62: [Double] = v219[1]
    let v105: Double = v62[5]
    return v105
  }
  func benchmark() {
    let v0: [[Double]] = [[0.0]]
    let v1: [[Double]] = [[1.0], [2.0], [3.0]]
    let v2: [[Double]] = [[4.0], [5.0], [6.0]]
    let v3: [[Double]] = [[7.0]]
    let v4: Double = 8.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 8.0
    for _ in 1...1000 {
      result = f0(v0, v1, v2, v3, v4)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen582.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  __attribute__((noinline))
  double f0(const std::vector<std::vector<double>> &v0, const std::vector<std::vector<double>> &v1, const std::vector<std::vector<double>> &v2, const std::vector<std::vector<double>> &v3, const double &v4) {
    double v9 = v4;
    v9 = v4;
    double v28 = v4;
    const std::vector<double> v22 { v9, v4, v28, v28, v9, v9 };
    std::vector<double> v16 = v22;
    const std::vector<double> v35 = v1[1];
    v16[2] = v9;
    const double v33 = v35[0];
    v16 = v22;
    v28 = v33;
    const std::vector<std::vector<double>> v95 { v22, v16, v22, v16, v22, v16 };
    std::vector<std::vector<double>> v219 = v95;
    const std::vector<double> v62 = v219[1];
    const double v105 = v62[5];
    return v105;
  }
  int main() {
    std::vector<std::vector<double>> v0({ { 0.0 } });
    std::vector<std::vector<double>> v1({ { 1.0 }, { 2.0 }, { 3.0 } });
    std::vector<std::vector<double>> v2({ { 4.0 }, { 5.0 }, { 6.0 } });
    std::vector<std::vector<double>> v3({ { 7.0 } });
    double v4(8.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1, v2, v3, v4);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }
